src/components/editor-page/editor-pane/linter/frontmatter-linter.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import type { Linter } from './linter'
+import type { EditorView } from '@codemirror/view'
+import type { Diagnostic } from '@codemirror/lint'
+import { extractFrontmatter } from '../../../../redux/note-details/frontmatter-extractor/extractor'
+import { load } from 'js-yaml'
+import type { RawNoteFrontmatter } from '../../../../redux/note-details/raw-note-frontmatter-parser/types'
+import { t } from 'i18next'
+
+export class FrontmatterLinter implements Linter {
+  lint(view: EditorView): Diagnostic[] {
+    const lines = view.state.doc.toString().split('\n')
+    const frontmatterExtraction = extractFrontmatter(lines)
+    if (!frontmatterExtraction.isPresent) {
+      return []
+    }
+    const frontmatterLines = lines.slice(0, frontmatterExtraction.lineOffset + 1)
+    try {
+      const rawNoteFrontmatter = load(frontmatterExtraction.rawText) as RawNoteFrontmatter
```

Comment:
The try-catch block makes this method very long but is only relevant for the yaml.load.
How about extracting the load call into a separate message and returning undefined on error?